### ISSUE TITLE
Structure Design

### ISSUE THREAD

**Issue by afong3** [temporalecologylab/TreeRings#7]
# Structure Design

## [OpenBuilds ACRO Kit](https://openbuildspartstore.com/openbuilds-acro-1010-40-x-40/) (~$700)

To not reinvent the wheel, an x-y gantry machine with a camera as the end tool seems to be a great option. After spending some time with the idea of designing the entire structure, I stumbled upon a DIY gantry kit supplier - OpenBuilds. They've made a kit that fits our specifications and for a comparable or less amount of money to designing our own! This includes planar x and y movement and 800 mm x 800 mm workspace.  Their kits come without an end tool. Where most of the customers would be adding a large router or laser engraver, we will be putting a camera. The kit has a movement resolution of 4.5 microns which is definitely enough for our application. 

Getting this kit would mean that no time would need to be spent on creating robust and precisely fitted brackets for motors, pulleys, or screws. I also think this would make it more accessible to labs that do not have the resources to manufacture parts themselves. 

Their mounting faces make it flexible to try out any camera system we want. The structure is designed to withstand loads from a router which is a comparable weight and size to a DSLR camera. And in the future, if we decide that adding a Z-axis is valuable, we can do that with a simple upgrade. 

Plus, their lead times are under a week. McMaster Carr is generally a few weeks. 

See below for what it looks like: 

![image](https://github.com/temporalecologylab/TreeRings/assets/62484502/af421e54-0708-41fe-9261-75054daad914)

## Necessary Components

| Component | Included in Kit | 
| --- | ---- | 
| Frame | ✅ |
| Stepper Motors | ✅ |
| Custom fitting plates | ✅ |
| Pulleys and Timing Belts | ✅ |
| Wiring Kit | ✅ (though we will likely need additional items as we introduce the camera) |
| Assembly Hardware | ✅ |
| Controller Board | ❓(see Notes) |
| Power Supply | ✅  |
| Bracket for camera | ❌ |

## Notes

I think that we should buy their kit, _without_ their motor controller/computer. Having theirs could be nice to just get things working immediately, although our control of the machine will not be how the controller board was designed to be used. Also, we can save about $300 by just using an Arduino motor controller. This saved money can be repurposed to buying an Nvidia Jetson edge computer instead of a Raspberry Pi. They are effectively Raspberry Pi's that have a more powerful GPU and hardware acceleration for computer vision tasks (i.e. image stitching). A Jetson Nano might be powerful enough to stitch images at a reasonable speed (about $300). If it doesn't, there are more powerful options such as a Jetson Xavier that would be up to the task (around $1000 if supply chains are doing good). 

## Example Project w/ Documentation

Luckily for us, somebody has already made a custom end tool for an ACRO kit and posted [detailed documentation](https://www.instructables.com/ACRO-Openbuilds-Pen-Plotter-Arduino-With-GRBL-and-/) online. We can use this for inspiration if we get stuck.

**Comment by DeirdreLoughnan** [comment]
@afong3 I think that is a great option! 

Our project has many interesting challenges, so using an existing gantry and focusing our energy elsewhere would be good. OpenBuilds is also a great ecosystem to work in.

**Comment by lizzieinvancouver** [comment]
@afong3 Sounds great to me too. Feels like we can go ahead, but @soleil-nocturne123 since you're doing the motor, can you quick check that you agree?

**Comment by soleil-nocturne123** [comment]
That sounds good to me! I have just added the stepper motors they recommended and the accessories to the Excel sheet.

**Comment by lizzieinvancouver** [comment]
@afong3 @soleil-nocturne123 Great! Please order away. @soleil-nocturne123 please ask @DeirdreLoughnan for help using the lab corporate card to get ordering underway.

**Comment by afong3** [comment]
I'm happy to share that the mechanical assembly from OpenBuilds has been put together! They provided a great step-by-step assembly video to follow. Here's the full 1m x 1m assembly on my table. 

![openbuilds_mechanical_setup](https://github.com/temporalecologylab/TreeRings/assets/62484502/3b647db2-deef-4a69-8b84-c1c9e86daa67)

**Comment by DeirdreLoughnan** [comment]
@afong3 That looks awesome! I am looking forward to seeing it in action!

**Comment by lizzieinvancouver** [comment]
@afong3 Very awesome! Now we just need to get a table for the lab.

**Comment by afong3** [comment]
Alright, the ACRO moves! 

![simple_movement_2](https://github.com/temporalecologylab/TreeRings/assets/62484502/31632064-8da0-466d-ad88-ac7d42d5b3d0)
![simple_movement](https://github.com/temporalecologylab/TreeRings/assets/62484502/c6d2c9b7-6d56-4866-b5c7-9eceaf513e31) 

We are missing some jumper caps (I temporarily improvised an alternative to see the setup move) which I just ordered.  The movement is a little bit jerky but this should be adjustable with how tight the wheels are fitting on the aluminum extrusions. I forgot my tools (so I ordered some for the lab) so I wasn't able to tune this today. 

Anyways, there are some more setup steps needed before precise movements are able to be made but this can wait until the jumpers arrive. 

Just thought I'd share!
